modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/logical/ExposeIndexRule.java

```diff
@@ -68,6 +68,9 @@ private static boolean preMatch(IgniteLogicalTableScan scan) {
         RexNode condition = scan.condition();
         ImmutableBitSet requiredCols = scan.requiredColumns();
 
+        if (igniteTable.isIndexRebuildInProgress())
```

Comment:
Calcite has some invariant and asserts preMatch after the rule is processed, so, in case of concurrent index rebuild flag change, there will be assertion error.